packages/utilities/src/merge.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Simple deep merge function. Takes all arguments are returns a new object which merges
+ * them all, in the order provided.
+ */
+export function merge<T = {}>(...args: Partial<T | null | undefined | false>[]): T {
+  // tslint:disable-next-line:no-any
+  const target: any = {};
+
+  for (const arg of args) {
+    _merge(target, arg);
+  }
+
+  return target as T;
+}
+
+function _merge<T>(target: T, source: T): T {
+  for (let name in source) {
+    if (source.hasOwnProperty(name)) {
+      const value = source[name];
+
+      if (typeof value === 'object') {
+        target[name] = merge(target[name], source[name]);
```

Comment:
if there's a cyclic reference in the object, this will crash

Reply:
If a cycle is detected, this should not recurse further and instead just treat the object reference as any value, i.e. `target[name]=source[name];`